engine/src/main/java/org/pentaho/di/job/entries/ftpsget/ftp4che/SecureDataFTPConnection.java

```diff
@@ -0,0 +1,146 @@
+/*! ******************************************************************************
+ *
+ * Pentaho Data Integration
+ *
+ * Copyright (C) 2018 by Hitachi Vantara : http://www.pentaho.com
+ *
+ *******************************************************************************
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+package org.pentaho.di.job.entries.ftpsget.ftp4che;
+
+import java.io.IOException;
+import java.net.InetSocketAddress;
+import java.util.List;
+
+import org.ftp4che.FTPConnection;
+import org.ftp4che.commands.Command;
+import org.ftp4che.commands.ListCommand;
+import org.ftp4che.exception.ConfigurationException;
+import org.ftp4che.exception.FtpIOException;
+import org.ftp4che.exception.FtpWorkflowException;
+import org.ftp4che.impl.SecureFTPConnection;
+import org.ftp4che.io.ReplyWorker;
+import org.ftp4che.io.SocketProvider;
+import org.ftp4che.reply.Reply;
+import org.ftp4che.util.ftpfile.FTPFile;
+
+public class SecureDataFTPConnection extends SecureFTPConnection {
+
+  /**
+   * SecureFTPConnection changed to force data channel encryption for implicit TLS/SSL connections with crypted data.
+   */
+  public SecureDataFTPConnection( FTPConnection conn, String password, int timeout ) throws ConfigurationException {
+    setConnectionType( conn.getConnectionType() );
+    setAddress( conn.getAddress() );
+    setUser( conn.getUser() );
+    setPassword( password );
+    setAccount( conn.getAccount() );
+    setPassiveMode( conn.isPassiveMode() );
+    setTryResume( conn.isTryResume() );
+    setProxy( conn.getProxy() );
+  }
+
+  /**
+   * This is the same as parent class except for also sending a PROT P command for the implicit modes with crypted data.
+   * {@inheritDoc}
+   */
+  public List<FTPFile> getDirectoryListing( String directory )
+    throws IOException, FtpWorkflowException, FtpIOException {
+
+    setConnectionStatusLock( CSL_INDIRECT_CALL );
+    setConnectionStatus( FTPConnection.BUSY );
+
+    ListCommand command = new ListCommand( directory );
+    SocketProvider provider = null;
+    String workDirectory = getWorkDirectory();
+
+    if ( isCryptedDataConnection() ) {
+      Command pbsz = new Command( Command.PBSZ, "0" );
+      ( sendCommand( pbsz ) ).dumpReply();
```

Comment:
why the extra parenthesis? and couldn't we send the command, and then call super.getDirectoryListing(directory)?

Reply:
sending the command before sounds like a better idea..leave this on hold